wazo_webhookd/plugins/mobile/celery_tasks.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2019-2023 The Wazo Authors  (see the AUTHORS file)
+# SPDX-License-Identifier: GPL-3.0-or-later
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING, Literal, overload
+
+import requests
+from celery import Task
+from pkg_resources import EntryPoint
+
+from wazo_webhookd.celery import app
+from .schema import NotificationDict
+
+if TYPE_CHECKING:
+    from ...services.mobile.plugin import (
+        PushNotification,
+        Service as PushNotificationService,
+    )
+    from ...types import WebhookdConfigDict
+
+
+MOBILE_SERVICE_ENTRYPOINT = 'mobile = wazo_webhookd.services.mobile.plugin'
+
+logger = logging.getLogger(__name__)
+
+if TYPE_CHECKING:
+
+    @overload
+    def get_service_class(name: Literal['Service']) -> type[PushNotificationService]:
+        ...
+
+    @overload
+    def get_service_class(name: Literal['PushNotification']) -> type[PushNotification]:
+        ...
+
+
+def get_service_class(name: str) -> type[PushNotificationService | PushNotification]:
+    return EntryPoint.parse(f'{MOBILE_SERVICE_ENTRYPOINT}:{name}').resolve()
+
+
+@app.task(bind=True)
+def send_notification(
+    task: Task,
+    config: WebhookdConfigDict,
+    notification: NotificationDict,
+) -> bool:
+    service_class: type[PushNotificationService] = get_service_class('Service')
+    notification_class: type[PushNotification] = get_service_class('PushNotification')
+
+    logger.debug(
+        "Attempting to send notification with payload: %s (attempt %d)",
+        notification,
+        task.request.retries + 1,
+    )
+    try:
+        external_tokens, external_config, jwt = service_class.get_external_data(
+            config, notification['user_uuid']
+        )
+    except requests.HTTPError as e:
+        if e.response.status_code == 404:
+            logger.error(
+                'Cannot send notification as no authentication exists for mobile (%s)',
+                str(e),
+            )
+            return False
+        raise
+
+    push_notification = notification_class(
+        task,
+        config,
+        external_tokens,
+        external_config,
+        jwt,
+    )
+    response = push_notification._send_notification(
+        notification['notification_type'],
+        notification['title'],
+        notification['body'],
+        notification['extra'],
+    )
+    logger.debug('Push response: %s', response)
+    if 'success' in response:
+        return response['success'] == 1  # type: ignore[typeddict-item]
+    return not response.get('error')
```

Comment:
This logic is weird without more context.

If I remember correctly, `success` is only present in FCM response and `error` in APNS response?

Can we explicit this somehow, or encapsulate the responses in an additional standardized structure like:
```python
push_notification_response = {
  'success': True or False,
  'upstream_type': 'apns' or 'fcm',
  'upstream_response': apns_response or fcm_response,
}
```

Reply:
Yeah, it's really weird what the functions return. FCM returns a response directly from the library and APNS returns a custom json generated from different properties of a response object. They are very different and don't have anywhere near the same information. We definitely could. It would probably be even better if we changed the return values from the two methods instead. It's probably `_send_notification` that should be standardized though. It's not normal that the same method returns two completely different objects.

Reply:
Completely agree. Do you want to make the change in this PR or is it too out-of-scope?

Reply:
Sure! No, I don't think it's out of scope as changes were made to that method as well.

Reply:
Does that work for you?

Reply:
yes! Thank you very much :)